Code/Framework/AzToolsFramework/Tests/SpinBoxTests.cpp

```diff
@@ -293,31 +296,44 @@ namespace UnitTest
 
     TEST_F(SpinBoxFixture, SpinBoxCheckHighValueTruncatesCorrectly)
     {
-        QString value = setupTruncationTest("0.9999999");
+        QLocale testLocale{ QLocale() };
+        QString testString = "0" + QString(testLocale.decimalPoint()) + "9999999";
+        QString value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.999");
+        testString = "0" + QString(testLocale.decimalPoint()) + "999";
+        EXPECT_TRUE(value == testString);
     }
 
     TEST_F(SpinBoxFixture, SpinBoxCheckLowValueTruncatesCorrectly)
     {
-        QString value = setupTruncationTest("0.0000001");
+        QLocale testLocale{ QLocale() };
+        QString testString = "0" + QString(testLocale.decimalPoint()) + "0000001";
+        QString value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.0");
+        testString = "0" + QString(testLocale.decimalPoint()) + "0";
+        EXPECT_TRUE(value == testString);
     }
 
     TEST_F(SpinBoxFixture, SpinBoxCheckBugValuesTruncatesCorrectly)
     {
-        QString value = setupTruncationTest("0.12395");
+        QLocale testLocale{ QLocale() };
+        QString testString = "0" + QString(testLocale.decimalPoint()) + "12395";
+        QString value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.123");
+        testString = "0" + QString(testLocale.decimalPoint()) + "123";
+        EXPECT_TRUE(value == testString);
 
-        value = setupTruncationTest("0.94496");
+        testString = "0" + QString(testLocale.decimalPoint()) + "94496";
+        value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.944");
+        testString = "0" + QString(testLocale.decimalPoint()) + "944";
+        EXPECT_TRUE(value == testString);
 
-        value = setupTruncationTest("0.0009999");
+        testString = "0" + QString(testLocale.decimalPoint()) + "0009999";
+        value = setupTruncationTest(testString);
 
-        EXPECT_TRUE(value == "0.0");
+        testString = "0" + QString(testLocale.decimalPoint()) + "0";
+        EXPECT_TRUE(value == testString);
```

Comment:
Repeat: We should replace the `EXPECT_TRUE` calls with `EXPECT_STREQ` to provide error messages

Reply:
EXPECT_TRUE for string comparisons can fail in debug. EXPECT_STREQ works, or using strcomp and checking the result is 0 works with EXPECT_TRUE.